httplib.h

```diff
@@ -6470,7 +6484,15 @@ inline bool Server::listen_internal() {
 #ifndef _WIN32
       }
 #endif
+
+#if defined _WIN32
+      // sockets conneced via WASAccept inherit flags NO_HANDLE_INHERIT, OVERLAPPED
+      socket_t sock = WSAAccept(svr_sock_, nullptr, nullptr, nullptr, 0);
+#elif defined __linux__
```

Comment:
I am not comfortable with using `__linux__`. If it possible to use `SOCK_CLOEXEC` instead?